### ISSUE TITLE
Folding does not work perfectly with Qiskit

### ISSUE THREAD

**Issue by yhindy** [unitaryfund/mitiq#209]
Hey everyone, 

I've been working on using mitiq to mitigate errors in a Qiskit VQE implementation, and have found that there are some issues with folding. Attached is an image of the scaled QP (on top) and the original QP.  If you take a look at the RY(2.2245) gate in the original, it happens before the controlled X on (3,0). When it is scaled, however, the RY(-2.2245) happens before the controlled X and the RY(2.2245) to undo that comes after the controlled X.

In other words, the moments are being violated. I think this can be fixed with qiskits `.barrier()` function. 

In any case, the bug leads to the wrong ansatz being computed, and the mitigated value cannot be used to do optimization. 

![Screen Shot 2020-06-16 at 3 56 08 PM](https://user-images.githubusercontent.com/11757328/84836296-5450a200-afea-11ea-9079-b455a6fd9f18.png)
